text/0000-shell-free-profile-d.md

```diff
@@ -0,0 +1,58 @@
+# Meta
+[meta]: #meta
+- Name: Shell-free Profile.d
+- Start Date: 2020-07-26
+- Author(s): @sclevine
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+Currently, `<layer>/profile.d/` scripts must be text files containing Bash 3+ scripts. This RFC proposes that we support arbitrary executables in `<layer>/profile.d/` that do not depend on a shell, but can still modify the environment of the app process.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, stacks must have Bash 3+ to support fully-dynamic modification of the environment at launch (via profile.d scripts).
+This means makes it difficult to modify the environment of app images that use minimal run base images.
+
+# What it is
+[what-it-is]: #what-it-is
+
+- Executable files in `<layer>/profile.d/` should be executed directly instead of sourced by Bash 3+.
```

Comment:
I vote for `/exec.d` or similar. I think the open question around shebang-less executable files pulls me in favor of this being more explicit.

Reply:
Done!